### PR TITLE
Code cleanup & modernization

### PR DESCRIPTION
**Pull request by zdimension**
Linting, cleaning, use of more recent Java features.

Should be squashed before merging.

### CODE REVIEW

**Comment by MarcinOrlowski** [comment]
I did quick review of some changes and my main complain is that this PR is massive -> 529 files changed. That'd pass BUT...
PR is too wide in scope (and is touching multiple subjects that could be handled separately). I'd be far more manageable if you'd split this PR by each type of changes separately (i.e. separate PR for cleaning up interfaces, separate PR for switching if to switch, separate PR for other things. In fact you could do "chained" PRs, where you branch each set of changes from your previous branch and then mark each PR with "DEPENDS_ON XXX" where XXX would be your parent PR. That the order of merge could be preserved and the diff would have more sense too.

So I am sorry but I am voting to reject this PR in current form as this is not reviewable really.

**Review by MarcinOrlowski — DISMISSED** [review]
I did quick review of some changes and my main complain is that this PR is massive -> 529 files changed. That'd pass BUT...
PR is too wide in scope (and is touching multiple subjects that could be handled separately). I'd be far more manageable if you'd split this PR by each type of changes separately (i.e. separate PR for cleaning up interfaces, separate PR for switching if to switch, separate PR for other things. In fact you could do "chained" PRs, where you branch each set of changes from your previous branch and then mark each PR with "DEPENDS_ON XXX" where XXX would be your parent PR. That the order of merge could be preserved and the diff would have more sense too.

So I am sorry but I am voting to reject this PR in current form as this is not reviewable really.

File: .github/CODEOWNERS
Comment (MarcinOrlowski) [inline .github/CODEOWNERS:5]: What the purpose for this change?

File: src/main/java/com/cburch/logisim/soc/pio/SocPio.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/soc/pio/SocPio.java:39]: Don't see much added value in this reformatting.

File: src/main/java/com/cburch/logisim/soc/rv32im/Rv32imRiscV.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/soc/rv32im/Rv32imRiscV.java:46]: Don't see much added value in this reformatting.

File: src/main/java/com/cburch/logisim/soc/util/AbstractExecutionUnitWithLabelSupport.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/soc/util/AbstractExecutionUnitWithLabelSupport.java:18]: Don't see much added value in empty lines between member function declarations.

File: src/main/java/com/cburch/logisim/soc/util/AssemblerExecutionInterface.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/soc/util/AssemblerExecutionInterface.java:34]: Don't see much added value in empty lines between member function declarations.

File: src/main/java/com/cburch/logisim/soc/vga/SocVga.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/soc/vga/SocVga.java:40]: Don't see much added value for this reformatting.

File: src/main/java/com/cburch/logisim/std/Builtin.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/Builtin.java:38]: Don't see much added value for this reformatting.

File: src/main/java/com/cburch/logisim/std/arith/Adder.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/arith/Adder.java:38]: Don't see much added value for this reformatting.

File: src/main/java/com/cburch/logisim/std/arith/ArithmeticLibrary.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/arith/ArithmeticLibrary.java:24]: Don't see much added value for this reformatting.

File: src/main/java/com/cburch/logisim/std/arith/BitFinder.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/arith/BitFinder.java:37]: Don't see much added value for this reformatting.

File: src/main/java/com/cburch/logisim/std/arith/Comparator.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/arith/Comparator.java:37]: Don't see much added value for this reformatting.

File: src/main/java/com/cburch/logisim/std/arith/Divider.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/arith/Divider.java:38]: Don't see much added value for this reformatting.

File: src/main/java/com/cburch/logisim/std/arith/FpAdder.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/arith/FpAdder.java:33]: Don't see much added value for this reformatting.

File: src/main/java/com/cburch/logisim/std/arith/FpComparator.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/arith/FpComparator.java:35]: Don't see much added value for this reformatting.

File: src/main/java/com/cburch/logisim/std/arith/FpDivider.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/arith/FpDivider.java:33]: Don't see much added value for this reformatting.

File: .github/CODEOWNERS
Comment (maehne) [reply .github/CODEOWNERS:5]: @zdimension has made considerable contributions over the past year. I am therefore not opposed to this change and would leave the final decision to @BFH-ktt1.

**Review by maehne — APPROVED** [review]
Thanks @zdimension for this considerable rework of the source code! Especially the use of the new switch case syntax helps with the readability. Other changes are of more cosmetic nature. Especially when it comes to line wrapping and indentation, personal preferences might diverge. Maybe, we should consider adopting an automatic source code formatting tool with a suitable configuration file to impose a certain level of consistency.

@zdimension: My review comments regarding the formatting are opinionated. They are meant as suggestions for you to decide whether you would like to take them into account or leave the formatting as proposed in your PR. Maybe, it would help the discussion/review by others if you just explain at suitable places the choice you made.

The length of this PR translated into a considerable effort to review it. Please, consider breaking it up into smaller chunks when reasonable, i.e., focussing on one topic at a time. I reckon that especially code style PRs tend to get long and contain quickly more than fixes to one aspect. I prefer longer PRs fixing also these issues than the risk of forgetting about some issues by postponing them. Doing interactive adding of changes with git to enable topical commits helps to some degree to keep thins manageable.

@BFH-ktt1: Due to the extend of the changes, I recommend that you also have a look at the changes before merging this PR.

**Comment by zdimension** [comment]
@maehne Thank you for you review. I concur on the fact that large PRs like this one are hard to review, especially for code bases such as Logisim's where there is a large volume of legacy code written before modern development tools. This one is pretty much a continuation of other large PRs I made last year in the same vein. The SonarCloud integration I set up a few months ago, and the formatting check in the CI are useful tools to have here in my opinion since they can help detect code smells and formatting issues.

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/HdlGeneratorFactory.java
Comment (zdimension) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/HdlGeneratorFactory.java:30]: Fields in an interface are already public, static and final, the modifiers are redundant (as is `public` in front of an interface method)

File: src/main/java/com/cburch/logisim/fpga/data/IoComponentTypes.java
Comment (zdimension) [reply src/main/java/com/cburch/logisim/fpga/data/IoComponentTypes.java:361]: `int * float` is of `float` type in Java, so the cast is redundant

**Review by maehne — DISMISSED** [review]
@zdimension: Thanks for all your efforts to unify the coding style and modernise the Logisim-evolution code base!

File: .github/CODEOWNERS
Comment (MarcinOrlowski) [reply .github/CODEOWNERS:5]: That's fine with me, but I am asking as this is solely for auto-assigning code reviewers of newly created PR. If  @zdimension is willing to review PRs then sure, go ahead as these can sometime stack up a small pile, otherwise I'd suggest not to.

Comment (zdimension) [reply .github/CODEOWNERS:5]: I am willing to, this is why I proposed this change to the CODEOWNERS file.

Comment (MarcinOrlowski) [reply .github/CODEOWNERS:5]: Perfect, thanks for clarification.

**Comment by zdimension** [comment]
@maehne Could you re-approve the PR, since I've made changes on the code bits you reviewed, and a review is required to allow merging? Thanks!

**Comment by maehne** [comment]
@zdimension: Sorry, but this and next week, it will be difficult for me to do code reviews due to the ongoing exam session. I will try to get to the reviews once things calm down a bit.

File: src/main/java/com/cburch/logisim/util/LineBuffer.java
Comment (R3dst0ne) [inline src/main/java/com/cburch/logisim/util/LineBuffer.java:498]: Please use a more descriptive variable name.
e.g. `line`

Comment (MarcinOrlowski) [reply src/main/java/com/cburch/logisim/util/LineBuffer.java:498]: Also I believe `String` can be `var`'ed. Best if could also be `final`.

**Review by BFH-ktt1 — APPROVED** [review]

File: src/main/java/com/cburch/logisim/util/LineBuffer.java
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/util/LineBuffer.java:498]: Please make a seperate PR for this one

Comment (R3dst0ne) [reply src/main/java/com/cburch/logisim/util/LineBuffer.java:498]: PR #1509

**Pull request merged**
